pkg/cli/admin/catalog/mirror.go

```diff
@@ -585,6 +590,20 @@ func aggregateICSPs(icsps [][]byte) []byte {
 	return aggregation
 }
 
+func (o *MirrorCatalogOptions) PostRun() error {
+	// If we have NOT set --path, the TempDir is set to true and we will delete
+	// Temporary folder.
+	if o.TempDir {
+		dir := strings.Split(o.IndexPath, ":")
```

Comment:
I find it always helps to be defensive and check the length of a slice before directly accessing an index. The assumption that `TempDir` equates to `len(dir) > 1` may not always hold.

Alternatively, you could encapsulate the assumption in a named type; e.g. `type DirPath string` that has a `TempDir() string` method, make `MirrorCatalogOptions.IndexPath` of that type, and call `tmp := o.IndexPath.TempDir()` here.

Reply:
Would the:
```go
if err := os.RemoveAll(dir[1]); err != nil {
```
Not already be catching this? Maybe I'm misunderstanding.

Reply:
How does something like this look to you?

```go
dir := strings.Split(o.IndexPath, ":")
if len(dir) != 1 {
    return fmt.Errorf("could not find dir to cleanup")
}
```